apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -141,7 +141,23 @@ export function parsePnpmDependencyKey(dependencyName: string, dependencyKey: st
   }
 
   if (!semver.valid(parsedVersionPart)) {
-    return undefined;
+    // Test for urls:
+    // Examples:
+    //     github.com/abc/def/188ed64efd5218beda276e02f2277bf3a6b745b2
+    //     github.com.au/abc/def/188ed64efd5218beda276e02f2277bf3a6b745b2
+    //     bitbucket.com/abc/def/188ed64efd5218beda276e02f2277bf3a6b745b2
+    //     bitbucket.co.in/abc/def/188ed64efd5218beda276e02f2277bf3a6b745b2
+    if (/^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}\/([^\/\\]+\/?)*([^\/\\]+)$/i.test(dependencyKey)) {
+      try {
+        const dependencySpecifier: DependencySpecifier = new DependencySpecifier(dependencyName, dependencyKey);
+        return dependencySpecifier;
+      }
+      catch (e) {
+        return undefined;
```

Comment:
This looks like an "unexceptional exception" -- is it expected for `DependencySpecifier` to throw an exception?  If so, shouldn't we be able to detect that and handle it properly?

Reply:
You are right, I think I can get rid of this try-catch.